app-arch/rpm/rpm-4.20.0.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CMAKE_MAKEFILE_GENERATOR=emake
+LUA_COMPAT=( lua5-{3,4} )
+PYTHON_COMPAT=( python3_{10..13} )
+
+inherit cmake lua-single python-single-r1 toolchain-funcs
+
+DESCRIPTION="The RPM Package Manager"
+HOMEPAGE="https://rpm.org/ https://github.com/rpm-software-management/rpm"
+SRC_URI="https://ftp.osuosl.org/pub/rpm/releases/rpm-$(ver_cut 1-2).x/${P}.tar.bz2"
+
+LICENSE="GPL-2 LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+IUSE="acl audit caps berkdb doc dbus nls openmp python readline selinux +sequoia +sqlite"
+REQUIRED_USE="
+	${LUA_REQUIRED_USE}
+	python? ( ${PYTHON_REQUIRED_USE} )
+"
+# Tests run against a Fedora container.
+RESTRICT="test"
+
+DEPEND="
+	${LUA_DEPS}
+	>=app-arch/bzip2-1.0.1
+	app-arch/libarchive:=
+	app-arch/xz-utils
+	app-arch/zstd:=
+	>=app-crypt/gnupg-1.2
+	>=dev-lang/perl-5.8.8
+	dev-libs/elfutils
+	>=dev-libs/popt-1.7
+	sys-apps/file
+	>=sys-libs/zlib-1.2.3-r1
+	virtual/libiconv
+	acl? ( virtual/acl )
+	audit? ( sys-process/audit )
+	caps? ( >=sys-libs/libcap-2.0 )
+	dbus? ( sys-apps/dbus )
+	nls? ( virtual/libintl )
+	python? ( ${PYTHON_DEPS} )
+	readline? ( sys-libs/readline:= )
+	selinux? ( sys-libs/libselinux )
+	sequoia? ( app-crypt/rpm-sequoia )
+	!sequoia? ( dev-libs/libgcrypt:= )
+	sqlite? ( dev-db/sqlite:3 )
+"
+BDEPEND="
+	virtual/pkgconfig
+	doc? ( app-text/doxygen )
+	nls? ( sys-devel/gettext )
+"
+RDEPEND="
+	${DEPEND}
+	selinux? ( sec-policy/selinux-rpm )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-4.19.1.1-musl-compat.patch
+	"${FILESDIR}"/${P}-libdir.patch
+)
+
+pkg_pretend() {
+	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
+}
+
+pkg_setup() {
+	lua-single_pkg_setup
+
+	use python && python-single-r1_pkg_setup
+
+	# bug #779769
+	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DENABLE_TESTSUITE=OFF
+		-DWITH_FAPOLICYD=OFF
+		-DENABLE_BDB_RO=$(usex berkdb)
+		-DENABLE_NLS=$(usex nls)
+		-DENABLE_OPENMP=$(usex openmp)
+		-DENABLE_PYTHON=$(usex python)
+		-DENABLE_SQLITE=$(usex sqlite)
+		-DWITH_ACL=$(usex acl)
+		-DWITH_AUDIT=$(usex audit)
+		-DWITH_CAP=$(usex caps)
+		-DWITH_DBUS=$(usex dbus)
+		-DWITH_SELINUX=$(usex selinux)
+		-DWITH_SEQUOIA=$(usex sequoia)
+		-DWITH_READLINE=$(usex readline)
+		-DWITH_DOXYGEN=$(usex doc)
+	)
```

Comment:
Could you add the other build options here, like `zstd`, `lzma` and so? I'd even hardcode some available options to always be enabled, like `WITH_LIBELF` just to show them in the ebuild.

Stuff like `ENABLE_ASAN` and `ENABLE_UBSAN` could probably go behind `debug` use flag. And `WITH_OPENSSL` could be hardcoded as `OFF` but visible in the ebuild.

https://github.com/rpm-software-management/rpm/blob/master/CMakeLists.txt

Reply:
Thanks for the review.
Just to confirm, **if** `ENABLE_ASAN` only added a build option (-fsanitize=address), there wouldn't be a need to add a USE flag, right?

Reply:
I added USE=sanitize anyway, since rpm does actually change the code path.
https://github.com/rpm-software-management/rpm/blob/rpm-4.20.0-release/lib/rpmplugins.c#L60

Reply:
Can you copy what I do in botan or elfutils instead please?

Reply:
Fixed `ENABLE_ASAN`.

Removed `ENABLE_UBSAN` since the only thing it does is add `-fsanitize=undefined` and `-fno-omit-frame-pointer`.